src/transformers/trainer.py

```diff
@@ -1536,11 +1539,11 @@ def _save(self, output_dir: Optional[str] = None):
         # They can then be reloaded using `from_pretrained()`
         if not isinstance(self.model, PreTrainedModel):
             if isinstance(unwrap_model(self.model), PreTrainedModel):
-                unwrap_model(self.model).save_pretrained(output_dir, _save_function=None)
+                unwrap_model(self.model).save_pretrained(output_dir, state_dict=self.model.state_dict())
```

Comment:
The model was saved after the test before (you can see line 1546 changed indentation) so the behavior is the same (it's actually even better since the keys that should be dropped from the state dict will now be dropped in `save_pretrained`.